<launch>
    <param name="/use_sim_time" value="false" />
    <arg name="scan_topic" default="scan" />
    <arg name="gt_odom_topic" default="base_gt_odom" />

    <arg name="rvizconfig" default="$(find Vive_mapper)/rviz/vive_mapper_exp.rviz"/>
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />

    <!-- Launch Vive_mapper node -->
    <node pkg="Vive_mapper" type="Vive_mapper" name="Vive_mapper" output="screen">
        <!-- Set parameters for mapping -->
        <param name="resolution" value="0.025" />
        <param name="width" value="1200" />
        <param name="height" value="1000" />
        <param name="occ_update_prob" value="0.7" />
        <param name="free_update_prob" value="0.4" />
        <param name="occ_threshold" value="0.95" />
        <param name="free_threshold" value="0.15" />
        <param name="stationary_distance_threshold" value="0.025" />
        <param name="stationary_angle_threshold" value="0.017453" />
        <param name="range_used_max" value="20.0" />
        <!--<param name="range_used_max" value="10.0" />-->
        <!--<param name="range_used_max" value="5.0" />-->
        <param name="base_frame_id" value="base_link" />
        <param name="Lidar_frame_id" value="laser" />
        <!-- Optional remapping -->
        <remap from="scan" to="$(arg scan_topic)"/>
        <remap from="base_gt_odom" to="$(arg gt_odom_topic)"/>
    </node>
    <!--
    <node pkg="husky_navigation" type="gt_2D" name="gt_2D" output="screen">
        <param name="use_carto_map" value="false"/>
    </node>
    -->
</launch>